admin_guide/enabling_EFK.adoc

```diff
@@ -0,0 +1,357 @@
+topic shows you how to deploy the stack using a template.
+
+Three components make up the EFK logging stack:
+
+* Elasticsearch: an object store where all logs are fed.
+* Fluentd: gathers logs from nodes and feeds them to Elasticsearch.
+* Kibana: a web UI for Elasticsearch.
+
+== Preperation for Deploying the EFK Stack
+
+Deploying the EFK stack requires:
+
+* A 'logging-deployer' secret
+* A 'logging-deployer' service account with the secret and create privileges
+* Sufficient volumes for ElasticSearch cluster storage
+* Optionally, a key and certificate for the Kibana server
+
+The deployer generates the objects necessary for communication within the
+cluster and defines secrets and templates to implement aggregated logging.
+
+. Create a new project. Once implemented in a single project, the EFK stack will apply to every project within your OpenShift cluster:
++
+----
+$ oc new-project <project_name>
+----
+
+. Create a link:../dev_guide/secrets.html[secret] for the deployer. The secret can be configured for your
+instance, but specific for the deployer.
++
+A set of files can be supplied in the secret:
++
+[cols="2",options="header"]
+|===
+|File Name
+|Description
+
+|*_kibana.crt_*
+|A browser-facing certificate for the Kibana server.
+
+|*_kibana.key_*
+|A key to be used with the Kibana certificate.
+
+|*_kibana-ops.crt_*
+|A browser-facing certificate for the Ops Kibana server.
+
+|*_kibana-ops.key_*
+|A key to be used with the Ops Kibana certificate.
+
+|*_server-tls.json_*
+|JSON TLS options to override the Kibana server defaults. Refer to NodeJS docs for available options and the default options for an example.
+
+|*_ca.crt_*
+|A certificate for a CA that will be used to sign all certificates generated by the deployer.
+
+|*_ca.key_*
+|A matching CA key.
+|===
++
+You can supply the files when creating a new secret:
++
+----
+$ oc secrets new logging-deployer \
+   kibana.crt=/path/to/cert kibana.key=/path/to/key
+----
+
+. Create the deployer link:../admin_guide/service_accounts.html[service account]:
++
+====
+----
+oc create -f - <<API
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: logging-deployer
+secrets:
+- name: logging-deployer
+API
+----
+====
++
+Then, ensure the logging-deployer can edit the service:
++
+====
+----
+$ oc policy add-role-to-user edit \
+          system:serviceaccount:logging:logging-deployer
+----
+====
+
+== Deploying the EFK Stack
+
+Deploy the entire EFK stack link:../dev_guide/templates.html[using a template],
+ensuring all future project configurations will feature the EFK stack. The
+default repository and branch of the template are set to origin/master. Change
+these settings to suit your own instance if desired. The template is found at
+the following GitHub repository:
+
+https://github.com/openshift/origin-aggregated-logging/blob/master/hack/templates/dev-builds.yaml
+
+[IMPORTANT]
+====
+The openshift-auth-proxy depends on a "node" base image, which is intended to be
```

Comment:
There's no need to mention this anyway, as we don't need base images because we don't need to build the images, they're supplied. All the image building stuff is for people building from github.